### PR TITLE
Update SSL options for OpenSSL v3

### LINKED ISSUE #112: Workaround for Heroku's 22 stack's `Faraday::SSLError (SSL_CTX_load_verify_file: system lib)`
It's a little beyond my understanding exactly what's going on but I have a work around for Heroku 22's stack producing a `Faraday::SSLError`.

# tl;dr
Set the client's SSL `ca_file` config variable to `/usr/lib/ssl/certs/ca-certificates.crt`* or `nil`. I do it in my client initializer and a simple one liner is
```ruby
IEX::Api.config.ca_file = '/usr/lib/ssl/certs/ca-certificates.crt'
```
\* from [Faraday's Wiki's SSL entry](https://github.com/lostisland/faraday/wiki/Setting-up-SSL-certificates#heroku-fedora-centos) for Heroku.

# What's going on?
The [client sets this value](https://github.com/dblock/iex-ruby-client/blob/f8d461cf00057f24ed4ce01109e05f7a1676ee1c/lib/iex/api/config/client.rb#L24) to `OpenSSL::X509::DEFAULT_CERT_FILE` which is `/usr/lib/ssl/cert.pem`. The file path, nor its lack of existence, does not change between Heroku's 20 and 22 stack, whereas OpenSSL is upgraded from `1.1.1f` to `3.0.2` (outlined in [Heroku's 22 Stack documentation](https://devcenter.heroku.com/articles/heroku-22-stack#openssl-3)). My assumption is that OpenSSL previously treated missing files and no file specified the same.

# Wrap it up
Ultimately I'm unsure what change to make to the library at this time but wanted to provide some information for the next person having trouble with their Heroku 22 stack update.

### PR DESCRIPTION
**Pull request by agrberg**
fixes #112

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
-### 1.6.1 (Next)
+### 2.0.0 (2022/07/20)
```
Comment (dblock) [inline CHANGELOG.md:1]: Change the date to Next, we haven't shipped anything.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 * [#111](https://github.com/dblock/iex-ruby-client/pull/111): Added support for fetching symbols for a region - [@mathu97](https://github.com/mathu97).
 * [#111](https://github.com/dblock/iex-ruby-client/pull/111): Added `Symbols#exchange_suffix`, `exchange_name`, `exchange_segment`, `exchange_segment_name`, and `lei` - [@mathu97](https://github.com/mathu97).
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md:4]: Put back your contribution here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+* [#113](https://github.com/dblock/iex-ruby-client/pull/113): Remove default SSL options for `ca_file` and `ca_path` as well as require Ruby >= 2.4 - [@agrberg](https://github.com/agrberg).
```
Comment (dblock) [inline CHANGELOG.md:4]: Either make a separate PR to get us up to Ruby >= 2.4 (or even 2.7), or at least make a separate line like this:

```
[#113](https://github.com/dblock/iex-ruby-client/pull/113): Drop support for Ruby < 2.4 ...
```

File: UPGRADING.md [hunk 4]
```diff
@@ -1,6 +1,20 @@
...
+If you previously relied on `OpenSSL::X509::DEFAULT_CERT_FILE` or `OpenSSL::X509::DEFAULT_CERT_DIR` to set these values you must now do so explicitly. E.g.:
+
+```ruby
+IEX::Api.configure do |config|
+  config.ca_file = OpenSSL::X509::DEFAULT_CERT_FILE
+  config.ca_path = OpenSSL::X509::DEFAULT_CERT_DIR
+end
+```
+
+\*At this time Ruby 2.7 is the earliest non-EOL branch. To maintain maximum compatibility 2.4 was chosen as `transform_values` was already being used and first appears in 2.4.
```
Comment (dblock) [inline UPGRADING.md:16]: Make two subsections under upgrading to >= 2.0, one for SSL cert options and another for Ruby minimum. Just say that the minimum ruby version is X, no need for a long explanation.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good! Needs some cleanup here and there.

File: CHANGELOG.md (hunk 3 above)
Comment (agrberg) [reply CHANGELOG.md]: As `transform_values` was already being used and the update to explicitly require 2.4 was two lines (one in gemspec + one in rubocop) I felt a separate PR wasn't valuable but did pull out the updates in the MD files.

**Comment by dblock** [comment]
Ooof I realized I merged it but CI didn't run. Care to take care of https://github.com/dblock/iex-ruby-client/issues/116 first?
